### PR TITLE
faq article on image pull issues

### PR DESCRIPTION
**Pull request by markxnelson**
This PR adds a new FAQ article about image pull issues to the documentation.
No tests - documentation only change.

### CODE REVIEW

File: docs-source/content/faq/cannot-pull-image.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+date: 2019-03-23T08:08:19-04:00
+draft: false
+---
+
+> My domain will not start and I see errors like `ImagePullBackoff` or `Cannot pull image`
+
+When you see these kinds of errors it means that Kubernetes cannot find your Docker image.
+The most common causes are: 
+
+* The `image` value is your domain resource is set incorrectly, meaning Kubernetes will be
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: The `image` value is your domain resource-> The `image` value in your domain resource

File: docs-source/content/faq/cannot-pull-image.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+When you see these kinds of errors it means that Kubernetes cannot find your Docker image.
+The most common causes are: 
+
+* The `image` value is your domain resource is set incorrectly, meaning Kubernetes will be 
+  trying to pull the wrong image
+* The image requires authentication or permission in order to pull it and you have not 
+  configured Kubernetes with the necessary credentials, for example in an `imagePullSecret`
+* You built the image on a machine that is not where your `kubelet` is running and Kubernetes
+  cannot see the image, meaning you need to copy the image to the worker nodes or put it in 
+  a Docker registry that is accessible the to worker nodes
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: that is accessible the to worker nodes-> that is accessible to the worker node

File: docs-source/content/faq/cannot-pull-image.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+Let's review what happens when Kubernetes starts a pod.
+
+![Pulling an image](/weblogic-kubernetes-operator/images/image-pull.png)
+
+The definition of the pod contains a list of container specifications.  Each container 
+specificiation contains the name (and optionally tag) of the image that should be used
+to run that container.  In the example above, there is a container called `c1` which is
+configured to use the docker image `some.registry.com/owner/domain1:1.0`.  This image
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: docker-> Docker (globally, when referring to the product)

File: docs-source/content/faq/cannot-pull-image.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+registry is `some.registry.com`, then owner is `owner`, the image name is `domain` 
+and the tag is `1.0`.  Tags are a lot like version numbers, but they are not required 
+to be numbers or to be in any particular sequence or format.  If you omit the tag, it 
+is assumed to be `latest`.
+
+{{% notice note %}}
+The docker tag `latest` is confusing - it does not actually mean the latest version of 
+the image that was created or published in the registry, it just literally means 
+whichever version the owner decided to call "latest".  Docker and Kubernetes make 
+some assumptions about latest, and it generally recommended to avoid using it and instead
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: and it generally recommended-> and it is generally recommended

File: docs-source/content/faq/cannot-pull-image.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+specify the actual version/tag that you really want. 
+{{% /notice %}}
+
+First, Kuberentes will check to see if the requested image is available in the local
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: Kuberentes-> Kubernetes

File: docs-source/content/faq/cannot-pull-image.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        --docker-server=some.registry.com \
+        --docker-username=bob \
+        --docker-password=bigSecret \
+        --docker-email=bob@some.com \
+        --namespace=default
+```
+
+In this command, you would replace `secret1` with the name of the secret; the `docker-server`
+is set to the registry name, without the `https://` prefix; the `docker-username`, `docker-password`
+and `docker-email` are set to match the credentials you use the authenticate to the remote
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: credentials you use the authenticate to-> credentials you use to authenticate to

File: docs-source/content/faq/cannot-pull-image.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        --docker-password=bigSecret \
+        --docker-email=bob@some.com \
+        --namespace=default
+```
+
+In this command, you would replace `secret1` with the name of the secret; the `docker-server`
+is set to the registry name, without the `https://` prefix; the `docker-username`, `docker-password`
+and `docker-email` are set to match the credentials you use the authenticate to the remote 
+Docker registry; and the `namespace` must be set to the same namespace where you intend to
+use the image
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: image->image.

File: docs-source/content/faq/cannot-pull-image.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+{{% notice note %}}
+Some registries may need a suffix making the `docker-server` something like `some.registry.com/v2`
+for example.  You will need to check with your registry provider's documentation to see if this is needed.
+{{% /notice %}}
+
+Once the secret is created, you need to tell Kubernetes to use it.  This is done by adding
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: Once the secret is created->After the secret is created
(once – don’t use to mean after; use only to mean less than twice.)

File: docs-source/content/faq/cannot-pull-image.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+{{% notice note }}
+You can provide mutliple `imagePullSecrets` if you need to pull Docker images from multiple 
+remote Docker registries or if your images require different authentication credentials.
+{{% /notice %}}
+
+#### Manually copying the image to your worker nodes
+
+If you are not able to use a remote Docker registry, for example if you Kubernetes cluster is
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: for example if you Kubernetes cluster->for example if your Kubernetes cluster

File: docs-source/content/faq/cannot-pull-image.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+Then copy that tar file to each worker node in your Kubernetes cluster and run this command
+on each node:
+
+```
+docker load < domain1.tar
+```
+
+#### Restart pods to clear the error
+
+After you have ensured the images are accessible on all worker nodes, you may need to restart
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: have ensured the images->have ensured that the images

File: docs-source/content/faq/cannot-pull-image.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+to run that container.  In the example above, there is a container called `c1` which is
+configured to use the docker image `some.registry.com/owner/domain1:1.0`.  This image
+name is in the format `registry address / owner / name : tag`, so in this case the 
+registry is `some.registry.com`, then owner is `owner`, the image name is `domain` 
+and the tag is `1.0`.  Tags are a lot like version numbers, but they are not required 
+to be numbers or to be in any particular sequence or format.  If you omit the tag, it 
+is assumed to be `latest`.
+
+{{% notice note %}}
+The docker tag `latest` is confusing - it does not actually mean the latest version of
```
Comment (rosemarymarano) [inline docs-source/content/faq/cannot-pull-image.md]: The docker tag-> The Docker tag

**Comment by markxnelson** [comment]
thanks @rosemarymarano  i made the updates. 
@rjeberhard ready to merge i believe

**Pull request merged**